components/opponent/wikis/trackmania/opponent_display_custom.lua

```diff
@@ -0,0 +1,198 @@
+---
+-- @Liquipedia
+-- wiki=trackmania
+-- page=Module:OpponentDisplay/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Class = require('Module:Class')
+local DisplayUtil = require('Module:DisplayUtil')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Table = require('Module:Table')
+
+local Opponent = Lua.import('Module:Opponent', {requireDevIfEnabled = true})
+local OpponentDisplay = Lua.import('Module:OpponentDisplay', {requireDevIfEnabled = true})
+local PlayerDisplay = Lua.import('Module:Player/Display', {requireDevIfEnabled = true})
+
+local OpponentDisplayCustom = Table.deepCopy(OpponentDisplay)
+
+local SCORE_STATUS = 'S'
+local NO_SCORE = -1
+local ZERO_SCORE = 0
+
+--[[
+Display component for an opponent entry appearing in a bracket match.
+]]
+OpponentDisplayCustom.BracketOpponentEntry = Class.new(
+	function(self, opponent, options)
+		self.content = mw.html.create('div'):addClass('brkts-opponent-entry-left')
+
+		if opponent.type == Opponent.team then
+			self:createTeam(opponent.template or 'tbd', options)
+		elseif opponent.type == Opponent.solo or opponent.type == Opponent.duo then
+			self:createPlayers(opponent)
+		elseif opponent.type == Opponent.literal then
+			self:createLiteral(opponent.name or '')
+		end
+
+		self.root = mw.html.create('div'):addClass('brkts-opponent-entry')
+			:node(self.content)
+	end
+)
+
+OpponentDisplayCustom.BracketOpponentEntry.createTeam = OpponentDisplay.BracketOpponentEntry.createTeam
+
+function OpponentDisplayCustom.BracketOpponentEntry:createPlayers(opponent)
+	local players = opponent.players
+	if #players == 1 then
```

Comment:
Wouldn't it be better to look at the Type?

Reply:
It lets us adding trio for example if we need it in the future, so I let it like this for later but we could just look at the type because now we only have Solo or Duo

Reply:
Looking to me, the control statement and the code scoped within looks like it's assuming it's 1 player. I assume this as a check for Solo?
then would do the job?
```suggestion
	if opponent.type == Opponent.solo
```
Or what am I missing?